tests/garage/torch/modules/test_gaussian_mlp_module.py

```diff
@@ -0,0 +1,362 @@
+def test_exp_min_std(input_dim, output_dim, hidden_sizes):
+    min_value = 10.
+
+    module = GaussianMLPModule(
+        input_dim=input_dim,
+        output_dim=output_dim,
+        hidden_sizes=hidden_sizes,
+        init_std=1.,
+        min_std=min_value,
+        std_share_network=False,
+        adaptive_std=False,
+        hidden_nonlinearity=None,
+        std_parameterization='exp',
+        hidden_w_init=nn.init.zeros_,
+        output_w_init=nn.init.zeros_
+    )
+
+    dist = module(torch.ones(5))
+
+    exp_variance = min_value**2
+
+    assert dist.variance.equal(torch.full((output_dim, ), exp_variance))
+
+
+@pytest.mark.parametrize("input_dim, output_dim, hidden_sizes", plain_settings)
+def test_exp_max_std(input_dim, output_dim, hidden_sizes):
+    max_value = 1.
+
+    module = GaussianMLPModule(
+        input_dim=input_dim,
+        output_dim=output_dim,
+        hidden_sizes=hidden_sizes,
+        init_std=10.,
+        max_std=max_value,
+        std_share_network=False,
+        adaptive_std=False,
+        hidden_nonlinearity=None,
+        std_parameterization='exp',
+        hidden_w_init=nn.init.zeros_,
+        output_w_init=nn.init.zeros_
+    )
+
+    dist = module(torch.ones(5))
+
+    exp_variance = max_value**2
+
+    assert dist.variance.equal(torch.full((output_dim, ), exp_variance))
+
+
+@pytest.mark.parametrize("input_dim, output_dim, hidden_sizes", plain_settings)
+def test_softplus_min_std(input_dim, output_dim, hidden_sizes):
+    min_value = 2.
+
+    module = GaussianMLPModule(
+        input_dim=input_dim,
+        output_dim=output_dim,
+        hidden_sizes=hidden_sizes,
+        init_std=1.,
+        min_std=min_value,
+        std_share_network=False,
+        adaptive_std=False,
+        hidden_nonlinearity=None,
+        std_parameterization='softplus',
+        hidden_w_init=nn.init.zeros_,
+        output_w_init=nn.init.zeros_
+    )
+
+    dist = module(torch.ones(5))
+
+    exp_variance = min_value**2
+
+    assert dist.variance.equal(torch.full((output_dim, ), exp_variance))
+
+
+@pytest.mark.parametrize("input_dim, output_dim, hidden_sizes", plain_settings)
+def test_softplus_max_std(input_dim, output_dim, hidden_sizes):
+    max_value = 1.
+
+    module = GaussianMLPModule(
+        input_dim=input_dim,
+        output_dim=output_dim,
+        hidden_sizes=hidden_sizes,
+        init_std=10,
+        max_std=max_value,
+        std_share_network=False,
+        adaptive_std=False,
+        hidden_nonlinearity=None,
+        std_parameterization='softplus',
+        hidden_w_init=nn.init.ones_,
+        output_w_init=nn.init.ones_
+    )
+
+    dist = module(torch.ones(5))
+
+    exp_variance = max_value ** 2
+
+    assert dist.variance.equal(torch.full((output_dim, ), exp_variance))
+
+
+@pytest.mark.parametrize("input_dim, output_dim, hidden_sizes", plain_settings)
+def test_std_network_output_values_pickable(input_dim, output_dim, hidden_sizes):
```

Comment:
these pickleable tests may not be necessary if pickling is provided by the torch API.

@krzentner what do you think?

Reply:
I generally don't care if there are "extra" tests, as long as they don't seem excessive.
Having said that, this test *should* be redundant.

Basically, I don't care, but please don't spend time writing these tests, since they're not needed.

Reply:
you mean there are too many test data in this test cases?
or this is not needed because PyTorch provides pickling?

Reply:
This one test just isn't needed because PyTorch provides the picking support.

Reply:
Oh.. ok then can I delete other pickling testcases?
I think I have one more pickling testcase in gaussian mlp policy

Reply:
Sure.

Reply:
delete every pickable tests